### ISSUE TITLE
[CLI] Formatting plugin does not work with `--config` and `--auto-correct` flags

### ISSUE THREAD

**Issue by sjjctl** [detekt/detekt#4929]
## Behavior

The following command should print off this warning,
which is associated with the formatting, and also
actually apply the formatting changes. It does not.

You have to call it _without_ the `-c` flag for that.

`java -jar $DETEKT_CLI --parallel -c .detekt.yml --auto-correct --plugins $DETEKT_FMT`

```text
SLF4J: Failed to load class "org.slf4j.impl.StaticLoggerBinder".
SLF4J: Defaulting to no-operation (NOP) logger implementation
SLF4J: See http://www.slf4j.org/codes.html#StaticLoggerBinder for further details.
WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by org.jetbrains.kotlin.com.intellij.util.ReflectionUtil (file:/home/shane/.local/lib/detekt-cli-1.20.0-all.jar) to field java.lang.Throwable.backtrace
WARNING: Please consider reporting this to the maintainers of org.jetbrains.kotlin.com.intellij.util.ReflectionUtil
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release
```

## Steps to Reproduce

Have some code that needs to be formatted.

Do a `git diff` before and after running this command

`java -jar $DETEKT_CLI --parallel -c .detekt.yml --auto-correct --plugins $DETEKT_FMT`

vs. this one

`java -jar $DETEKT_CLI --parallel --auto-correct --plugins $DETEKT_FMT`

Only the second, without the config flag, will actually format.

## Context

Want to do both linting and formatting in one step in CI/CD

## Your Environment

* Version of detekt used: 1.20.0
* Operating System and version: Ubuntu 20.04

**Comment by sjjctl** [comment]
Yes, the name is intentional.

The config is practically the default with a few modifications, we have not prettified it. So you can do a `diff --color` with the default generated.

[detekt.txt](https://github.com/detekt/detekt/files/8853897/detekt.txt)

**Comment by cortinico** [comment]
Can you try again by including this https://github.com/detekt/detekt/blob/main/detekt-formatting/src/main/resources/config/config.yml in your final config.yml?

**Comment by sjjctl** [comment]
Yes, i just pasted them at the bottom.

Everything works now.

I'm not sure what this means in terms of the default config and your documentation. But that solved it for me.

I have some more questions about improving the formatting to be more aggressive, like IDEA. I'll try to get some notes together for a new issue.

I was also thinking the default config should, if it's okay to update it, be either prettified or yamllint compliant? Currently it has some serious syntax warnings

**Comment by cortinico** [comment]
Closing as it's solved.

> I'm not sure what this means in terms of the default config and your documentation. But that solved it for me.

How have you generated the config?

> be either prettified or yamllint compliant? Currently it has some serious syntax warnings

We generate the config on the fly, so if it's not yamllint complient, we shoudl update the generator to produce valid output (i.e. yes tha'ts a bug).

**Comment by sjjctl** [comment]
I just used `detekt-cli -gc` with 1.20.0. Afaik, it does not include formatting rules or that section out of the box.

**Comment by BraisGabin** [comment]
> I have some more questions about improving the formatting to be more aggressive, like IDEA. I'll try to get some notes together for a new issue.

Out rules are just a wrapper around ktlint. If you want ktlint to be more "aggresive" you can open an issue there but I think that they are happy with their "agressiveness". I think that what you are looking for is [ktfmt](https://facebookincubator.github.io/ktfmt/).

**Comment by sjjctl** [comment]
I would be in favor of adding a note to the `-gc` flag in output text for help. Once someone generates the config, they are unlikely to figure out on their own that the config generation was (or could be) influenced by plugins.

ktfmt looks promising. Unfortunately running it with the `--kotlinlang-style` results in a mass format. Our project uses Kotlin style guide (with a few tweaks in `.idea/codeStyles/Project.xml`). If we are switching to a new tool, it ideally has full compatibility with the IntelliJ command line formatter and our existing options. The IDEA command line formatter is a bit cumbersome, and only one instance at a time can be run per machine. But more of a mild inconvenience, than truly crippling.

And the editorconfig here: https://github.com/facebookincubator/ktfmt/blob/main/docs/editorconfig/.editorconfig-kotlinlang

Looks much different from ours, ours is very simple only 5-8 `ij_*` props.  Maybe they can make it more configurable, with a `Project.xml` even.
